openquake/hazardlib/gsim/idriss_2014.py

```diff
@@ -0,0 +1,183 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Module exports :class:`Idriss2014`,
+"""
+from __future__ import division
+
+import numpy as np
+
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, SA
+
+
+class Idriss2014(GMPE):
+    """
+    Implements GMPE developed by Idriss 2014 and published as "An NGA-West2
+    Empirical Model for Estimating the Horizontal Spectral Values Generated
+    by Shallow Crustal Earthquakes.
+    (2014, Earthquake Spectra, Volume 30, No. 3, pages 1155 - 1177).
+
+    Idriss (2014) defines the GMPE only for the case in which Vs30 >= 450 m/s.
+    In the present implementation no check is made for the use of this model
+    for sites with Vs30 < 450 m/s
+    """
+    #: Supported tectonic region type is active shallow crust
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.ACTIVE_SHALLOW_CRUST
+
+    #: Supported intensity measure types are spectral acceleration,
+    #:and peak ground acceleration
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        SA
+    ])
+
+    #: Supported intensity measure component is orientation-independent
+    #: measure :attr:`~openquake.hazardlib.const.IMC.RotD50`
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.RotD50
+
+    #: Supported standard deviation types are total
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+    ])
+
+    #: Required site parameters is Vs30
+    REQUIRES_SITES_PARAMETERS = set(('vs30', ))
+
+    #: Required rupture parameters are magnitude, and rake.
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'rake'))
+
+    #: Required distance measure is Rrup
+    REQUIRES_DISTANCES = set(('rrup', ))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # extracting dictionary of coefficients specific to required
+        # intensity measure type.
+        C = self.COEFFS[imt]
+
+        mean = (self._get_magnitude_scaling_term(C, rup.mag) +
+                self._get_distance_scaling_term(C, rup.mag, dists.rrup) +
+                self._get_style_of_faulting_term(C, rup.rake) +
+                self._get_site_scaling_term(C, sites.vs30))
+
+        stddevs = self._get_stddevs(imt,
+                                    rup.mag,
+                                    len(dists.rrup),
+                                    stddev_types)
+        return mean, stddevs
+
+    def _get_magnitude_scaling_term(self, C, mag):
```

Comment:
Since self.COEFF is a class attribute, it would be actually cleaner to avoid passing 'C' from the params, and instead add 'imt' as a method parameter and call inside the method 
C = self.COEFFS[imt]